File: 卡码/代码随想录/JS/回溯/01-组合/LeetCode_0216.js
//216. 组合总和 III
const combinationSum3 = function(k, n) {
  const path=new Array();
  const res=new Array();
  const backtracking=(k,n,sum,index)=>{
    if(sum>n) return;
    if(k==path.length){
      if(sum==n) res.push(Array.from(path));
      return;
    }
    for(let i=index;i<=10-(k-path.length);i++){
      path.push(i);
      backtracking(k,n,sum+i,i+1);
      path.pop();
    }
  }
  backtracking(k,n,0,1);
  return res;
};

File: 卡码/代码随想录/Python/回溯/01-组合/LeetCode_0216.py
#216. 组合总和 III
from typing import *

class Solution:
  def __init__(self):
    self.path:List[int]=[]
    self.res:List[List[int]]=[]
  def backtracking(self,k:int,n:int,sum:int,index:int):
    if sum>n:
      return
    if k==len(self.path):
      if sum==n:
        self.res.append(self.path[:])
      return
    for i in range(index,11-(k-len(self.path))):
      self.path.append(i)
      self.backtracking(k,n,sum+i,i+1)
      self.path.pop()
  def combinationSum3(self, k: int, n: int) -> List[List[int]]:
    self.backtracking(k,n,0,1)
    return self.res